Move dotnet-scaffold telemetry article to docs/core/tools/ (#55211)
dotnet/docs: docs/core/tools/dotnet-scaffold-telemetry.md

* File an issue in the [dotnet/scaffolding](https://github.com/dotnet/scaffolding/issues) repository.
* Send an email to [dotnet@microsoft.com](mailto:dotnet@microsoft.com) for investigation.

The telemetry feature collects the following data for .NET SDK version 8.0 and later:

* Timestamp of invocation.
* Three-octet IP address used to determine the geographical location.
* Operating system and version.
* Runtime ID (RID) the tool is running on.
* Whether the tool is running in a container.
* Hashed Media Access Control (MAC) address: a cryptographically (SHA256) hashed and unique ID for a machine.
* Kernel version.
* dotnet-scaffold version.
* Hashed tool information (tool name, tool version, tool package name, tool package version, chosen scaffolder category, related scaffolding categories).
* Tool level (global or local tool).
* Hashed command invoked (for example, `mvccontroller`) and whether it succeeded.
* dotnet-scaffold-aspnet scaffolder name, step names, and whether they succeeded.
* dotnet-scaffold-aspire scaffolder name and whether it succeeded.
* dotnet-scaffold-aspnet scaffolder validation method name and whether it succeeded.

> * dotnet-scaffold-aspire scaffolder validation method name and whether it succeeded.

Reviewer: ```suggestion
The telemetry feature collects the following data for .NET SDK version 8.0 and later:

* Timestamp of invocation.
* Three-octet IP address used to determine the geographical location.
* Operating system and version.
* Runtime ID (RID) the tool is running on.
* Whether the tool is running in a container.
* Hashed Media Access Control (MAC) address: a cryptographically (SHA256) hashed and unique ID for a machine.
* Kernel version.
* dotnet-scaffold version.
* Hashed tool information (tool name, tool version, tool package name, tool package version, chosen scaffolder category, related scaffolding categories).
* Tool level (global or local tool).
* Hashed command invoked (for example, `mvccontroller`) and whether it succeeded.
* dotnet-scaffold-aspnet scaffolder name, step names, and whether they succeeded.
* dotnet-scaffold-aspire scaffolder name and whether it succeeded.
* dotnet-scaffold-aspnet scaffolder validation method name and whether it succeeded.
* dotnet-scaffold-aspire scaffolder validation method name and whether it succeeded.

The telemetry feature **does not** collect:

* Personal data, such as usernames, email addresses, or URLs.
* Any project data.

The data is sent securely to Microsoft servers and held under restricted access.

Protecting your privacy is important to us. If you suspect the telemetry feature is collecting sensitive data or the data is being insecurely or inappropriately handled, take one of the following actions:

* File an issue in the [dotnet/scaffolding](https://github.com/dotnet/scaffolding/issues) repository.
* Send an email to [dotnet@microsoft.com](mailto:dotnet@microsoft.com) for investigation.
```

Things seemed out of order to me. The H2 is "Data points" and then it immediately went into what it *didn't* collect and the privacy policy etc. So I put that at the end of the section instead.

---
title: dotnet-scaffold telemetry
description: Learn about the telemetry collected by the dotnet-scaffold CLI tool.
author: tdykstra
ms.author: tdykstra
ms.date: 08/03/2026
ai-usage: ai-assisted
---
# dotnet-scaffold telemetry

> The `dotnet-scaffold` command includes a telemetry feature that collects usage data. This feature helps the `dotnet-scaffold` team understand how the tool is used so it can be improved.

Reviewer 2: ```suggestion
The `dotnet-scaffold` command includes a telemetry feature that collects usage data. This feature helps the `dotnet-scaffold` team understand how the tool is used so they can improve it.
```

## How to opt out

The `dotnet-scaffold` telemetry feature is enabled by default. To opt out of the telemetry feature, set the `DOTNET_SCAFFOLD_TELEMETRY_OPTOUT` environment variable to `1` or `true`.

## Disclosure

> When you run the `dotnet-scaffold` tool the first time, it displays output similar to the following example. The text might vary slightly depending on the version of the tool you're running. This "first run" experience is how Microsoft notifies you about data collection.

Reviewer 2: ```suggestion
When you run the `dotnet-scaffold` tool for the first time, it displays output similar to the following example. The text might vary slightly depending on the version of the tool you're running. This "first run" experience is how Microsoft notifies you about data collection.
```

| .NET SDK versions | Data |
|--------------|------|
| >=8.0        | Timestamp of invocation. |
| >=8.0        | Three-octet IP address used to determine the geographical location. |
| >=8.0        | Operating system and version. |
| >=8.0        | Runtime ID (RID) the tool is running on. |
| >=8.0        | Whether the tool is running in a container. |
| >=8.0        | Hashed Media Access Control (MAC) address: a cryptographically (SHA256) hashed and unique ID for a machine. |
| >=8.0        | Kernel version. |
| >=8.0        | dotnet-scaffold version. |
| >=8.0        | Hashed tool information (tool name, tool version, tool package name, tool package version, chosen scaffolder category, related scaffolding categories). |
| >=8.0        | Tool level (global or local tool). |
| >=8.0        | Hashed command invoked (for example, `mvccontroller`) and whether it succeeded. |
| >=8.0        | dotnet-scaffold-aspnet scaffolder name, step names, and whether they succeeded. |
| >=8.0        | dotnet-scaffold-aspire scaffolder name and whether it succeeded. |
| >=8.0        | dotnet-scaffold aspnet scaffolder validation method name and whether it succeeded. |

> | >=8.0        | dotnet-scaffold aspire scaffolder validation method name and whether it succeeded. |

Reviewer 2: ```suggestion
| >=8.0        | dotnet-scaffold-aspnet scaffolder validation method name and whether it succeeded. |
| >=8.0        | dotnet-scaffold-aspire scaffolder validation method name and whether it succeeded. |
```

---
title: dotnet-scaffold telemetry
description: Learn about the telemetry collected by the dotnet-scaffold CLI tool.
author: tdykstra
ms.author: tdykstra

> ms.date: 11/12/2024

Reviewer: Update?

Reviewer 2: Updated, thanks!

ai-usage: ai-assisted

> uid: core/tools/dotnet-scaffold-telemetry

Reviewer: ```suggestion
```

---
# dotnet-scaffold telemetry

The `dotnet-scaffold` command includes a telemetry feature that collects usage data. This feature helps the `dotnet-scaffold` team understand how the tool is used so it can be improved.

## How to opt out

The `dotnet-scaffold` telemetry feature is enabled by default. To opt out of the telemetry feature, set the `DOTNET_SCAFFOLD_TELEMETRY_OPTOUT` environment variable to `1` or `true`.

## Disclosure

> When you run the `dotnet-scaffold` tool the first time, it displays output similar to the following example. The text may vary slightly depending on the version of the tool you're running. This "first run" experience is how Microsoft notifies you about data collection.

Reviewer: ```suggestion
When you run the `dotnet-scaffold` tool the first time, it displays output similar to the following example. The text might vary slightly depending on the version of the tool you're running. This "first run" experience is how Microsoft notifies you about data collection.
```

Read more about dotnet-scaffold telemetry:
https://aka.ms/dotnet-scaffold/telemetry
Read more about .NET CLI Tools telemetry:
https://aka.ms/dotnet-cli-telemetry
```

To suppress the "first run" experience text, set the `DOTNET_SCAFFOLD_SKIP_FIRST_TIME_EXPERIENCE` environment variable to `1` or `true`.

## Data points

The telemetry feature doesn't collect:

* Personal data, such as usernames, email addresses, or URLs.
* Any project data.

The data is sent securely to Microsoft servers and held under restricted access.

Protecting your privacy is important to us. If you suspect the telemetry feature is collecting sensitive data or the data is being insecurely or inappropriately handled, take one of the following actions:

* File an issue in the [dotnet/scaffolding](https://github.com/dotnet/scaffolding/issues) repository.
* Send an email to [dotnet@microsoft.com](mailto:dotnet@microsoft.com) for investigation.

The telemetry feature collects the following data.

| .NET SDK versions | Data |
|--------------|------|

> | >=8.0        | Timestamp of invocation. |

Reviewer: Since the .NET SDK version doesn't vary at all, I would just state it up front and turn the table into bullet points.

Reviewer 2: Yes, good point.  Removing that column and making the version clear in the a statement above the table instead...

| >=8.0        | Three-octet IP address used to determine the geographical location. |
| >=8.0        | Operating system and version. |
| >=8.0        | Runtime ID (RID) the tool is running on. |
| >=8.0        | Whether the tool is running in a container. |
| >=8.0        | Hashed Media Access Control (MAC) address: a cryptographically (SHA256) hashed and unique ID for a machine. |
| >=8.0        | Kernel version. |
| >=8.0        | dotnet-scaffold version. |

> | >=8.0        | Hashed tool information (tool name, tool version, tool package name, tool package version, chosen scaffolder category, related scaffolding categories) |

Reviewer 2: ```suggestion
| >=8.0        | Hashed tool information (tool name, tool version, tool package name, tool package version, chosen scaffolder category, related scaffolding categories). |
```

> | >=8.0        | Tool level (global or local tool) |

Reviewer 2: ```suggestion
| >=8.0        | Tool level (global or local tool). |
```

| >=8.0        | Hashed command invoked (for example, `mvccontroller`) and whether it succeeded. |

> | >=8.0        | dotnet-scaffold-aspnet scaffolder name, step names, and whether they succeeded. |

Reviewer: I see dotnet-scaffold-aspnet on line 62 and without the last hypen on line 64. Are they supposed to be different?

Reviewer 2: Good catch. both `dotnet-scaffold-aspnet` and `dotnet-scaffold-aspire` should have a hyphen. Fixed. Thanks!

| >=8.0        | dotnet-scaffold-aspire scaffolder name and whether it succeeded. |
| >=8.0        | dotnet-scaffold aspnet scaffolder validation method name and whether it succeed. |

> | >=8.0        | dotnet-scaffold aspire scaffolder validation method name and whether it succeed. |

Reviewer: ```suggestion
| >=8.0        | dotnet-scaffold aspnet scaffolder validation method name and whether it succeeded. |
| >=8.0        | dotnet-scaffold aspire scaffolder validation method name and whether it succeeded. |
```

> | >=8.0        | dotnet-scaffold aspire scaffolder validation method name and whether they succeed. |

Reviewer 2: ```suggestion
| >=8.0        | dotnet-scaffold aspnet scaffolder validation method name and whether it succeed. |
| >=8.0        | dotnet-scaffold aspire scaffolder validation method name and whether it succeed. |
```

---
title: dotnet-scaffold telemetry
description: Learn about the telemetry collected by the dotnet-scaffold CLI tool.
author: tdykstra
ms.author: tdykstra
ms.date: 08/03/2026
ai-usage: ai-assisted
---
# dotnet-scaffold telemetry

The `dotnet-scaffold` command includes a telemetry feature that collects usage data. This feature helps the `dotnet-scaffold` team understand how the tool is used so they can improve it.

> `dotnet-scaffold` is a .NET global tool that generates code to speed up development. The telemetry feature applies when you use the tool, which isn't included with the .NET SDK. To install it, run the following command:

Reviewer 2: ```suggestion
`dotnet-scaffold` is a .NET global tool that generates scaffolding code for common components, such as controllers, views, and pages, so you don't have to write that boilerplate by hand. The telemetry feature applies when you use the tool. `dotnet-scaffold` isn't included with the .NET SDK. To install it, run the following command:
```

```dotnetcli
dotnet tool install --global Microsoft.dotnet-scaffold
```

> For more information about the tool, see the [dotnet/Scaffolding repository](https://github.com/dotnet/Scaffolding) and the [Getting Started guide](https://github.com/dotnet/Scaffolding/blob/main/docs/Getting-Started.md).

Reviewer 2: ```suggestion
For more information about the `dotnet-scaffold` tool, see the [dotnet/Scaffolding repository](https://github.com/dotnet/Scaffolding) and the [Getting Started guide](https://github.com/dotnet/Scaffolding/blob/main/docs/Getting-Started.md).
```
